enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,634 @@
+haproxy.config at this [code location](https://github.com/openshift/router/blob/master/images/router/haproxy/conf/haproxy-config.template#L609).
+The router sets this variable based on the Route's
+`spec.httpDeleteHeaders` field value. ServiceAliasConfig will be populated at this [code location](https://github.com/openshift/router/blob/master/pkg/router/template/router.go#L927)
+
+Please refer the structure in point C) to understand the field DeleteHTTPResponseHeaders of struct ServiceAliasConfig.
+
+E) We will be creating an immutable/harcoded deny list of HTTP Response headers in the [library-go](https://github.com/openshift/library-go/blob/2a52b7ce52f99e2ed185718380b43734877f2439/pkg/route/validation/validation.go#L48-L141) 
+which won't be allowed to be set(replace)/append. Eg - HSTS.
+
+### Risks and Mitigations
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+- Would this enhacement cover setting request headers ?
+
+  Answer: As per [epic](https://issues.redhat.com/browse/RFE-464) the requirement for the customers seems to be for HTTP response headers only.
+  We need confirmation from the PM regarding this.
+
+- Would we be able to set load balancing alogorithm based on headers ?
+
+  Answer: No, this out of scope for this enhancement so it will be considered as a non-goal.
+
+- Would we be able to delete or append headers  ?
+
+  Answer: Yes, this enhancment will cover deletion and appending of HTTP Response Headers as it is a requirement of the epic.
+
+- Would some headers be blocked for setting, deleting and appending ?
+
+  Answer: Yes, this enhancement won't allow the user to set/replace, append or delete HSTS headers. We already a way to set this which is via route 
+  annotation. Providing one more way to add this can cause issues.
+
+- Would we be able to [Replace a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-a-header-by-using-a-regular-expression)?
+  Answer: No, this is out of scope for this enhancement.
+
+- Would we be able to [Replace part of a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-part-of-a-header-by-using-a-regular-expression) ?
+  Answer:  No, this is out of scope for this enhancement.
+
+### Test Plan
+
+The controller that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.
+
+The operator has end-to-end tests; for this enhancement, a test is added that
+(1) creates an IngressController that specifies certain response
+headers should be set(replace) or append or deleted., (2) sends an HTTP request to one of the standard
+routes (such as the console route), and (3) verifies that the expected headers
+are logged.
```

Comment:
Expected headers are logged? What does this mean exactly? How exactly we would check for the presence of the headers we set or the absence of the headers we delete?

Reply:
curl -vk <console route> shall show the headers set in response.